ui/src/app/pages/login/account-creation/account-creation-institution.spec.tsx

```diff
@@ -0,0 +1,151 @@
+import {createEmptyProfile} from 'app/pages/login/sign-in';
+import {AccountCreationInstitution, Props} from './account-creation-institution';
+import {ConfigApiStub} from 'testing/stubs/config-api-stub';
+import {InstitutionApiStub} from 'testing/stubs/institution-api-stub';
+import {registerApiClient} from 'app/services/swagger-fetch-clients';
+import defaultServerConfig from 'testing/default-server-config';
+import {institutionApi} from 'app/services/swagger-fetch-clients';
+import SpyInstance = jest.SpyInstance;
+import {Dropdown} from 'primereact/dropdown';
+import {waitOneTickAndUpdate} from 'testing/react-test-helpers';
+import {defaultInstitutions} from 'testing/stubs/institution-api-stub';
+import {InstitutionalRole} from 'generated/fetch';
+import {AccountCreationOptions} from 'app/pages/login/account-creation/account-creation-options';
+
+let mockGetPublicInstitutionDetails: SpyInstance;
+
+type AnyWrapper = (ShallowWrapper|ReactWrapper);
+
+let props: Props;
+function component(): ReactWrapper {
+  return mount(<AccountCreationInstitution {...props}/>);
+}
+
+function getInstitutionDropdown(wrapper: AnyWrapper): Dropdown {
+  return wrapper.find('Dropdown[data-test-id="institution-dropdown"]').instance() as Dropdown;
+}
+
+function getEmailInput(wrapper: AnyWrapper): AnyWrapper {
+  return wrapper.find('[data-test-id="contact-email"]').hostNodes();
+}
+
+function getRoleDropdown(wrapper: AnyWrapper): Dropdown {
+  return wrapper.find('Dropdown[data-test-id="role-dropdown"]').instance() as Dropdown;
+}
+
+function getSubmitButton(wrapper: AnyWrapper): AnyWrapper {
+  return wrapper.find('[data-test-id="submit-button"]');
+}
+
+
+beforeEach(() => {
+  serverConfigStore.next(defaultServerConfig);
+  registerApiClient(ConfigApi, new ConfigApiStub());
+  registerApiClient(InstitutionApi, new InstitutionApiStub());
+
+  props = {
+    profile: createEmptyProfile(true),
+    onComplete: (profile: Profile) => {},
+    onPreviousClick: (profile: Profile) => {}
+  };
+
+  mockGetPublicInstitutionDetails = jest.spyOn(institutionApi(), 'getPublicInstitutionDetails');
+});
+
+it('should render', async() => {
+  const wrapper = component();
+  expect(wrapper.exists()).toBeTruthy();
+});
+
+it('should load institutions list', async() => {
+  const wrapper = component();
+  await waitOneTickAndUpdate(wrapper);
+
+  expect(mockGetPublicInstitutionDetails).toHaveBeenCalled();
+
+  const options = getInstitutionDropdown(wrapper).props.options as Array<Object>;
+  expect(options.length).toEqual(defaultInstitutions.length);
+});
+
+it('should show user-facing error message on data load error', async() => {
+  mockGetPublicInstitutionDetails.mockRejectedValueOnce(new Response(null, {status: 500}));
+  const wrapper = component();
+  await waitOneTickAndUpdate(wrapper);
+
+  expect(wrapper.find('[data-test-id="data-load-error"]').exists).toBeTruthy();
+});
+
+const academicSpecificOption = AccountCreationOptions.institutionalRoleOptions.find(
```

Comment:
Nit: I might move these up to be with the other helper functions, because they kinda shook me being in the middle of the test cases.